src/Jenssegers/Mongodb/Eloquent/Model.php

```diff
@@ -239,20 +241,11 @@ public function originalIsEquivalent($key, $current)
             return false;
         }
 
-        if ($this->isDateAttribute($key)) {
-            $current = $current instanceof UTCDateTime ? $this->asDateTime($current) : $current;
-            $original = $original instanceof UTCDateTime ? $this->asDateTime($original) : $original;
-
-            return $current == $original;
-        }
-
-        if ($this->hasCast($key)) {
```

Comment:
The only problem I see is a casted attributes, probably we need to check after keys was casted back to their values.

Reply:
@divine  Looking at the code, I don't see it is required to check for casts. The input values passed by getDirty() are uncasted:
```php
    public function getDirty()
    {
        $dirty = [];

        foreach ($this->getAttributes() as $key => $value) {
            if (! $this->originalIsEquivalent($key, $value)) {
                $dirty[$key] = $value;
            }
        }

        return $dirty;
    }
    public function getAttributes()
    {
        return $this->attributes;
    }
```
Besides, I don't see why one should use Laravel <=6 casting system for MongoDB, I don't see a test for it in the repository either. Laravel 7 custom casts are a different story. I hope they won't make things go wrong. I've already had my custom cast implementation along side this PR in Larave 5.8 production without an issue.

Reply:
Well, we're dropping support for laravel 6 (no other way to drop object id auto-casting) and this branch will be focused for versions > 7.0, so that's the reason why I've asked this question.

This might be an issue with Laravel 7 when it wouldn't check dirtiness for casted objects.

Yes, there are no tests, I think a lot of tests are missing, might take a look myself (for tests).

Reply:
@divine Then I should send this PR on top of your own PR #1988 ?

Reply:
Probably no, it's not ready yet, still have something to do. I will let you know, ok?

Reply:
@divine I have already done some checks and coding for Laravel 7. Sounds like casting in Laravel 7 also doesn't break things.
I'll send you that PR anyway. I'll wait for you to let me work on my unset-field PR #1667 after this is done.